### PR TITLE
feat: `structlog` plugin & bug fixes

### PR DESCRIPTION
**Pull request by cofin**
This implements a `StructlogPlugin` class to make it easier to configure logging in a single place.

Fixes a bug when caused when calling `app.debug`.  In some cases, the structlog configuration will only be initalized lazily.  This abstraction does not have the `setLevel` function.

Details:
- Detect of `logger` has `setLevel` before calling
- Set even message name to be init-cap
- Add `set_level` interface to config 
- Allow structlog printer to detect if console is TTY enabled.  If so, use a Struglog color formatter with Rich traceback printer.
- Auto-configure stdlib logger to use the structlog logger
- Adds `StructlogPlugin`

### CODE REVIEW

**Comment by provinzkraut** [comment]
@cofin I think you need to rebase the changes onto `develop` :eyes:

File: litestar/plugins/structlog.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from litestar.cli._utils import console
```
Comment (provinzkraut) [inline litestar/plugins/structlog.py:6]: Console is only available when `rich` is installed, not sure this should be used here.

Comment (cofin) [reply litestar/plugins/structlog.py:6]: Isn't Rich included by default now?

Comment (provinzkraut) [reply litestar/plugins/structlog.py:6]: Oh. Yes. Forgot about that :grimacing:

**Comment by provinzkraut** [comment]
@cofin This shoud fix #2151 and #1672, right?

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**